backend/src/utils/errors.ts

```diff
@@ -0,0 +1,24 @@
+// src/utils/errors.ts
+class AppError extends Error {
+  public statusCode: number;
+
+  constructor(message: string, statusCode: number) {
+    super(message);
+    this.statusCode = statusCode;
+    Error.captureStackTrace(this, this.constructor);
+  }
+}
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2430019945>#d33549</a><span> - 10/22/2024, 07:15 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Consider adding a `name` property to the `AppError` class to distinguish between different types of errors. This can be helpful for error handling and logging.
</div>
<div id="fix">
<b>Fix:</b> Add a `name` property to the `AppError` class and set it in the constructor.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -2,9 +2,11 @@
 class AppError extends Error {
   public statusCode: number;
 +  public name: string;

   constructor(message: string, statusCode: number) {
     super(message);
     this.statusCode = statusCode;
 +    this.name = this.constructor.name;
     Error.captureStackTrace(this, this.constructor);
   }
 }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged